internal/pkg/service/buffer/storage/statistics/collector/collector.go

```diff
@@ -1,153 +1,211 @@
+// Package collector provides collecting and saving of the storage statistics.
 package collector
 
 import (
 	"context"
+	"sort"
 	"sync"
 	"time"
 
 	"github.com/benbjohnson/clock"
-	"github.com/c2h5oh/datasize"
-	etcd "go.etcd.io/etcd/client/v3"
 
 	"github.com/keboola/keboola-as-code/internal/pkg/log"
-	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/config"
-	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/store/key"
-	"github.com/keboola/keboola-as-code/internal/pkg/service/common/etcdop/serde"
-	"github.com/keboola/keboola-as-code/internal/pkg/service/common/servicectx"
-	"github.com/keboola/keboola-as-code/internal/pkg/service/common/utctime"
-	"github.com/keboola/keboola-as-code/internal/pkg/telemetry"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/storage"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/storage/level/local/writer"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/storage/statistics"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/storage/statistics/repository"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
 )
 
-// Collector collects node statistics in memory and periodically synchronizes them to the database.
+const (
+	DefaultSyncInterval = 1 * time.Second
+	DefaultSyncTimeout  = 30 * time.Second
+)
+
+// Collector collects writers statistics and saves them to the database.
+// Collection is triggered periodically and on the writer close.
 type Collector struct {
```

Comment:
Statistics collector je cely zrefactorovany, odporucam pozriet vysledny kod + test.

Dovod prepisania bol primarne v tom, ze teraz mame kompresiu `on-the-fly` uz pri zapise na disk, predtym sa robila az pri uploade.

Z toho dovodu si statistiky o zapise vedie priamo `writer.Writer` a `Collector` ich iba pravidelne a `OnClose` zbiera.